### ISSUE TITLE
Calendar widget - handling single date events that have a startTime AND endTime

### ISSUE THREAD

**Issue by troughc** [culturecreates/footlight-app#1182]
Problem: when a single date event has a start time AND end time, no time is displayed in the widget preview. Only the date is displayed. (single date events with start times DO display the start time in the widget preview)
Expected behaviour: Single date events with start AND end time should display the start time in the widget preview just like single date events with ONLY start times do.
### Notes
No change is required for multiple date events (event series) with times. These type of events are displayed in the widget the same way date range events are displayed (with no time included). This is acceptable. 
This can be done in between other more important issues (lower priority)

**Comment by troughc** [comment]
@SyamBabu-M do you have thoughts on this, how much work would it take? Should we schedule this for after the event series work is done?

**Comment by SyamBabu-M** [comment]
> [@SyamBabu-M](https://github.com/SyamBabu-M) do you have thoughts on this, how much work would it take? Should we schedule this for after the event series work is done?

I need to check with @AbhishekPAnil as he is working on the event series right now.

**Comment by SyamBabu-M** [comment]
@troughc Please test this in CMS staging.

**Comment by troughc** [comment]
TEsted in staging: PASSED, the time is showing up correctly thank you!
But, see comment below for a couple of small bugs

First: when I switch the widget language from French to english for Tout Culture and then try to leave the tab, I get an error for unsaved changes. However you cannot SAVE the widget settings explicitly. Expected behaviour is that you can change settings in the widget and leave the tab without seeing a save error. OR we have to add a save button to that tab (I need to discuss with Stephanie about whether we should go in this direction). 

Second: this is more of a question and  may or may not impact the widget. When I add an end time to an event, and the end time is on the next day, my event still shows up as a single date (which I think is fine). My question is, how does it look in the structured data? What is the start-date-time and end-date-time?
To check, please use [this event](https://staging.cms.footlight.io/dashboard/6308ef4a7f771f00431d939a/events/6629f5500796c3006456eadc)
This event starts in the evening at 8PM and ends the next day at 11AM.
I would also like to know what happens when an event starts at 8PM one evening and ends at 12AM the same night.

Thanks!

**Comment by SyamBabu-M** [comment]
> First: when I switch the widget language from French to english for Tout Culture and then try to leave the tab, I get an error for unsaved changes. However you cannot SAVE the widget settings explicitly. Expected behaviour is that you can change settings in the widget and leave the tab without seeing a save error.

I have made this change.

**Comment by SyamBabu-M** [comment]
> I would also like to know what happens when an event starts at 8PM one evening and ends at 12AM the same night.

We would have to add this as a multiday or custom event. If we select a single-day event then we can't add this as the end time 12:00 AM is before 8:00 PM.

> Second: this may or may not impact the widget. When I add an end time to an event, and the end time is on the next day, my event still shows up as a single date (which I think is okay).
> My question is, how does this look in the structured data? What is the start-date-time and end-date-time?
> To check, please use [this event](https://staging.cms.footlight.io/dashboard/6308ef4a7f771f00431d939a/events/6629f5500796c3006456eadc)
> This event starts in the evening at 8PM and ends the next day at 11AM.

This event does not have an end time. Also, I don't think I follow you about the end time being on the next day when its a single day event.
If a event start at 8 PM and ends at 11 am next day, then we will have to save is as a custom event with 2 occurrences.

**Comment by troughc** [comment]
> > First: when I switch the widget language from French to english for Tout Culture and then try to leave the tab, I get an error for unsaved changes. However you cannot SAVE the widget settings explicitly. Expected behaviour is that you can change settings in the widget and leave the tab without seeing a save error.
> 
> I have made this change. Please test it [here](https://pr-1330.dla06mbe0wyl7.amplifyapp.com/)

@SyamBabu-M I tested this and it looks good - no save message appears: PASSED
However, there is a small display bug in the test build - the list doesn't appear when i first go to the widget tab. Steps to reproduce:
1. Go to the widget tab for Tout Culture. English language is always selected. Notice that the listing on the page is empty (no results found.
2. show the preview popup - listing is displayed there. 
3. Close the preview popup - listing still not displayed on the page.
4. switch language to French. listing appears in french
5. Switch language to English. Listing now appears on the page.
Expected behavior: Listing results should be displayed at step 1 and all the way through.

Screenshots of step 1 and 2.
step 1
<img width="989" alt="Screenshot 2024-09-17 at 12 00 50 PM" src="https://github.com/user-attachments/assets/9ef6794c-303a-477d-ac48-b700d9314b17">


Step 2
<img width="1132" alt="Screenshot 2024-09-17 at 12 01 06 PM" src="https://github.com/user-attachments/assets/bfb164e9-1566-4826-bd01-5fca9c6894b3">

**Comment by SyamBabu-M** [comment]
@troughc I am not able to reproduce this bug after following the steps. When I go to the widget settings page, there is always data in the listing. I will keep an eye on this issue.

**Comment by troughc** [comment]
Still to do: handle events that start on one day and end on the next (in the middle of the night). They are single date events in the FE, but in the BE, the start and end time formats reflect actual date-times.
For example:
in the FE (CMS), a user can select a single date, then add a START time in the AM or PM (8AM or 8PM for example) and an END time happening later, maybe the next day in the AM (7AM for example). 
In the list view and in the event details, the event still is listed as a single date event, happening on the date selected (but with an end time happening afterwards, on the next day)
In the backend, the event is a single date event but the start time is on the date selected and the end time is on the next day

**Comment by SyamBabu-M** [comment]
@troughc I discussed this with the team. 
@AbhishekPAnil and @SuhailAliyar proposed checking the difference b/w startDateTime and endDateTime and if it is less than 24 hours then we will consider this as a single-day event and treat it as such. Currently, we are planning to go through with this implementation. 

@troughc What are your thoughts on this method?

**Comment by troughc** [comment]
### Recap of meeting with @troughc and @SyamBabu-M 
As discussed, there is no scenario where a user could create a single date (or multiple dates / event series) event that has an individual event that is longer than 24 hours. With these two date types each event could only be as long as 23 hours and 45 minutes in the UI (for example, from 7 AM - 6:45 AM the next day)
Only a date RANGE event (a single event that last longer than a day) can be longer than 24 hours. And Date range events do not allow users to put in a start time and end time. They can only select a start date and an end date.
### In process
@SyamBabu-M  is investigating a potential issue with multiple dates events (event series with sub events) that occurs when an single subevent goes overnight. And will make sure the CMS properly handles an event (or sub-event) that goes overnight. 
Thanks :)

**Comment by troughc** [comment]
@SyamBabu-M let me know if you would like to move the above outstanding work to a new issue and deploy this one - I realize that the outstandign work it is a bit of a departure from the original request in this issue.  Up to you!

**Comment by SyamBabu-M** [comment]
@troughc We can move this to new issue if the earlier fixed are urgent. Otherwise it ok to work on new requirement as a part of this issue.

**Comment by troughc** [comment]
@SyamBabu-M  let's look at this one together if possible tomorrow, I also created a test event for multidates in Tout Culture called [Caitlin Test](https://pr-1330.dla06mbe0wyl7.amplifyapp.com/dashboard/6308ef4a7f771f00431d939a/events/66f307262c9e6c00a1038ee3). We can review it together to make sure it works as intended in terms of how dates are displayed in the widget and in the CMS.


Also, I noticed in the widget display that Some of the dates seem wonky. For example, Passe Partout dates are listed in the wrong order - backwards (i didn't touch this event, I just noticed the issue) :
![Image](https://github.com/user-attachments/assets/565ed303-b82c-4dd8-accd-1c52ac8c7d18)

**Comment by troughc** [comment]
@SyamBabu-M as discussed I am going to write up a new issue with a user story describing how to handle events that go overnight. This will include a UI change suggestion. You do not need to do that work in this issue.

I did a final test and there is one issue (I think we may have lost sight of the expected behaviour in the widget)

Here is the original scenario and criteria to pass this issue:

_When a single date event has a start time AND end time, no time is displayed in the widget preview. Only the date is displayed. (single date events with start times DO display the start time in the widget preview)
**Expected behaviour:** Single date events that include start AND end time should display the start time in the widget preview just like single date events that only have start times do._

Tested in chrome: Did not pass
Problem: For events that have a start time AND end time, both start time and end time are displayed in the listing widget. 
Fix: **Only display the START time in the listing widget (not the end time).**
